### ISSUE TITLE
Rewrite the commits

### ISSUE THREAD

**Issue by jaraco** [pypa/distutils#263]
Gah. I spent the last couple of days merging PRs in distutils and had forgotten that the issue references needed to be made absolute. Now we've got a bunch of bare numbers that when merged into Setuptools will create nuisance links to the wrong issues.

It does sound like it's possible to reword commits, even merges, with git, so I'm going to try to rewrite a lot of history.

**Comment by jaraco** [comment]
The last merge to setuptools was 55982565e, so I'll be examining the commits since then.

**Comment by jaraco** [comment]
I issued the command: `git rebase -i 5598256 --rebase-merges`, which produced [this TODO file](https://gist.github.com/jaraco/3bcecc1d2685a4f93ae7f92fc196027f).

I then ran a substitution on it: `s/merge -C (.* pull request) #/merge -c \1 #/`. The lowercase `-c` signals to reword the commit.

Git started rewriting the commits allowing me to re-word the merge commits, but then ran into a merge conflict:

```
 distutils main @ git rebase -i 55982565e --rebase-merges
[detached HEAD 48bf47c5c] Merge pull request pypa/distutils#245 from DimitriPapadopoulos/FURB
 Date: Sat Apr 13 09:32:32 2024 -0400
[detached HEAD c6fad4c77] Merge pull request pypa/distutils#247 from pypa/hotfix/246-linker-args-list
 Date: Sat Apr 13 16:40:32 2024 -0400
[detached HEAD 5aff9d862] Merge pull request pypa/distutils#239 from DimitriPapadopoulos/extra_pairs_of_quotes
 Date: Sat Apr 13 20:16:59 2024 -0400
Auto-merging pyproject.toml
CONFLICT (content): Merge conflict in pyproject.toml
Could not apply 851e71af1... Merge-https-github-com-jaraco-skeleton # Merge https://github.com/jaraco/skeleton
```

That doesn't make any sense. No content has changed. How could there possibly be a merge conflict?

**Comment by jaraco** [comment]
I asked Gemini and [it had some ideas](https://gist.github.com/jaraco/3bcecc1d2685a4f93ae7f92fc196027f/1a4bfaadbd92a34226521f4ec8bf6547c8d41338). It seems that maybe `rerere` might be what I'm seeking. Let's try it.

**Comment by jaraco** [comment]
It's not obvious from the docs if `rerere` needs to be enabled when the conflict was originally resolved, in which case it won't be any help here since I resolved the conflict months ago.

**Comment by jaraco** [comment]
Nope. Even with `rerere` enabled, the same error occurs. It did record a pre-image, but I'll need to resolve the conflict manually. I thought I'd figured out a technique whereby I could `git checkout --ours .; git add .; git rebase --continue` when I ran into conflicts, but this didn't work. That ended up resolving some conflicts incorrectly.

And now I'm worried I've "recorded" these bad conflict resolutions with rerere.

I also tried doing a `git reset <original merge hash> .`, but that left the conflicts in place.

Next I tried doing `git checkout <original merge hash> .` and that seems to be doing the trick. It worked for all but one of the merge conflicts. There was one merge conflict where a file had been deleted on one side and changed on the other. I had to manually resolve the deletion. The [history is here](https://gist.github.com/jaraco/27a4680cea4516adba22f3fb3d0e5a86).

Ultimately it shows that the code is exactly the same at main, suggesting it's all the same at the commits too. For historical reference, I've pushed the old main to an archive ref:

```
 distutils main @ git push origin origin/main:refs/archive/263-reword
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)
To https://github.com/pypa/distutils
 * [new reference]       origin/main -> refs/archive/263-reword
```

And force-pushed the reworded commits to main as f3b225449.
